src/app/api/v1/endpoints/extract_data.py

```diff
@@ -168,7 +171,7 @@ def create_response(coord, srs):
     return None
 
 
-def extract_text(pdf_page: fitz.Page, user_defined_bbox: fitz.Rect) -> ExtractDataResponse:
+def extract_text(pdf_page: fitz.Page, user_defined_bbox: fitz.Rect) -> ExtractDataResponse | HTTPException:
```

Comment:
Can be reverted to just `ExtractDataResponse` as return type? Also, can't we be more specific and put `ExtractTextResponse` as return type?